test/cql-pytest/test_describe.py

```diff
@@ -2567,3 +2568,90 @@ def aux(row: DescRow) -> DescRow:
             assert restore_stmts == res
         finally:
             cql.execute(f"DROP KEYSPACE IF EXISTS {ks}")
+
+### ===========================================================================
+
+# `varchar` is an alias for `text`. Scylla defaults to `text` and we test it here.
+# It's not very important, but it's better to have a test that detects changes
+# like that and verifies we handle that case as expected.
+def test_restore_udf_with_varchar(cql, test_keyspace):
+    fn_content = """\
+        (arg {type_name})
+        RETURNS NULL ON NULL INPUT
+        RETURNS {type_name}
+        LANGUAGE lua
+        AS $$
+        return arg
+        $$;"""
+    # Get rid of indentation in the string.
+    fn_content = textwrap.dedent(fn_content)
+
+    with new_function(cql, test_keyspace, fn_content.format(type_name="varchar")) as fn:
+        udf_name = fn
+        udf_row = cql.execute(f"DESC FUNCTION {test_keyspace}.{fn}").one()
+        udf_stmt = udf_row.create_statement
+
+        expected_stmt = f"CREATE FUNCTION {test_keyspace}.{fn}{fn_content.format(type_name="text")}"
+        assert udf_stmt == expected_stmt
+
+    try:
+        cql.execute(udf_stmt)
```

Comment:
Now I understand why you called this test "restore" (I still find this name ambigous, but at least I understand now where it came from).
But.... What's the point of this execute??? One line above, you assert that udf_statement is a specific statement. What's the point of running this statement now? We already know it's correct.

Reply:
The problem we're solving in this PR has been present in Scylla for some time now. It would've been found out immediately if we tried executing the statements we get from describe. I don't think doing that for every test is that important, but since I wrote the tests as reproducers, I thought it would be good to do that. I'm fine with dropping the restore part, though

Reply:
To make a long story short, your tests all something like this:

1. Run statement S1 to create an object
2. Run DESC on the object to get a description S2
3. Assert that the strings S1 and S2 are equal
4. Run S2 to check it executes, and do nothing with the result.

Because of step 3, step 4 is completely pointless - you know S1=S2, so executing S2 is the same as executing S1, which you know works because you did it in step 1.

So I think removing step 4 will not this test weaker, but it will make it easier to understand and will allow you to rename it from a "restore" test to a "desc_function" test.

By the way, you're right that you could have designed tests that are based on DESCing an object, running the result, and then somehow comparing the original object and the second object. But it wouldn't be these tests. It's not what these tests are doing.

Reply:
That's a good point